src/edu/washington/escience/myria/api/encoding/DatasetStatus.java

```diff
@@ -21,16 +22,19 @@
    * @param numTuples The number of tuples in the dataset.
    * @param queryId The query that created this dataset.
    * @param created When this dataset was created, in ISO8601 format.
+   * @param howPartitioned How this dataset was partitioned.
    */
   @JsonCreator
   public DatasetStatus(@JsonProperty("relationKey") final RelationKey relationKey,
       @JsonProperty("schema") final Schema schema, @JsonProperty("numTuples") final long numTuples,
-      @JsonProperty("queryId") final long queryId, @JsonProperty("created") final String created) {
+      @JsonProperty("queryId") final long queryId, @JsonProperty("created") final String created,
+      @JsonProperty("howPartitioned") final PartitionFunction howPartitioned) {
```

Comment:
Yeah there could be a lot... will extend it to a dictionary but only add workers for this PR.